python/cudaq/kernel/ast_bridge.py

```diff
@@ -1673,6 +1673,25 @@ def bodyBuilder(iterVal):
                     is_adj=False)
                 return
 
+            # Handle the case where we are capturing an opaque kernel
+            # function. It has to be in the capture vars and it has to
+            # be a PyKernelDecorator.
+            if node.func.id in self.capturedVars and node.func.id not in globalKernelRegistry:
+                from .kernel_decorator import PyKernelDecorator
```

Comment:
nit / Q: import statement not at the top?

Reply:
If this import is at the top of the file you'll get a cyclic dependency upon `import cudaq`. We only need it locally here to check that a capture variable is an aliased kernel (capture variable of `PyKernelDecorator` type).